### ISSUE TITLE
Feature: move ArenaSpectator into a module and away from the core

### ISSUE THREAD

**Issue by Kitzunu** [azerothcore/azerothcore-wotlk#17683]
### Describe your feature request or suggestion in detail

move ArenaSpectator into a module and away from the core

(The addon for anyone interested https://github.com/azerothcore/arena-spectator)

**Comment by heyitsbench** [comment]
I've made a proof of concept branch showing what all would need to be removed to accomplish this. Can be seen [here](https://github.com/azerothcore/azerothcore-wotlk/compare/master...heyitsbench:azerothcore-wotlk:no_more_spectator). Also removed (I assume) an overworld spectate command as well, that may be worth keeping around as a GM tool at least.

**Comment by Kitzunu** [comment]
Now you can create the module :P

**Comment by Helias** [comment]
I do not know what the lua script does, but we already have a module for the arena spectator which is quite updated https://github.com/Gozzim/mod-npc-spectator

Shall we close this issue?

**Comment by heyitsbench** [comment]
> Shall we close this issue?

No, since the code is still in the core (not a Lua script, that's just the addon component) and technically does a different thing.
